convert2rhel/unit_tests/actions/report_test.py

```diff
@@ -1465,3 +1467,110 @@ def test_get_combined_results_and_message(results, expected):
     combined_results_and_message = report.get_combined_results_and_message(results)
 
     assert combined_results_and_message == expected
+
+
+@pytest.mark.parametrize(
+    ("actions", "expected"),
+    (
+        (
+            {
+                "action1": {
+                    "result": {"level": STATUS_CODE["ERROR"]},
+                    "messages": [{"level": STATUS_CODE["SUCCESS"]}],
+                },
+                "action2": {
+                    "result": {"level": STATUS_CODE["WARNING"]},
+                    "messages": [{"level": STATUS_CODE["SUCCESS"]}],
+                },
+            },
+            "ERROR",
+        ),
+        (
+            {
+                "action1": {
+                    "result": {"level": STATUS_CODE["SUCCESS"]},
+                    "messages": [],
+                },
+                "action2": {
+                    "result": {"level": STATUS_CODE["SUCCESS"]},
+                    "messages": [],
+                },
+            },
+            "SUCCESS",
+        ),
+        (
+            {
+                "action1": {
+                    "result": {"level": STATUS_CODE["SUCCESS"]},
+                    "messages": [{"level": STATUS_CODE["WARNING"]}],
+                },
+                "action2": {
+                    "result": {"level": STATUS_CODE["SUCCESS"]},
+                    "messages": [],
+                },
+            },
+            "WARNING",
+        ),
+        (
+            {
+                "action1": {
+                    "result": {"level": STATUS_CODE["INFO"]},
+                    "messages": [{"level": STATUS_CODE["SUCCESS"]}],
+                },
+                "action2": {
+                    "result": {"level": STATUS_CODE["SUCCESS"]},
+                    "messages": [],
+                },
+            },
+            "INFO",
+        ),
+        (
+            {
+                "action1": {
+                    "result": {"level": STATUS_CODE["INFO"]},
+                    "messages": [{"level": STATUS_CODE["SUCCESS"]}],
+                },
+                "action2": {
+                    "result": {"level": STATUS_CODE["SKIP"]},
+                    "messages": [],
+                },
+            },
+            "SKIP",
+        ),
+        (
+            {
+                "action1": {
+                    "result": {"level": STATUS_CODE["INFO"]},
+                    "messages": [{"level": STATUS_CODE["OVERRIDABLE"]}],
+                },
+                "action2": {
+                    "result": {"level": STATUS_CODE["ERROR"]},
+                    "messages": [],
+                },
+            },
+            "ERROR",
+        ),
+    ),
+)
+def test_find_highest_report_level_expected(actions, expected):
+    """Should be sorted descending from the highest status to the lower one."""
+    result = report.find_highest_report_level(actions)
+    assert result == expected
+
+
+def test_find_highest_report_level_unknown_status():
```

Comment:
Maybe I would use the `pytest.mark.parametrize` for easier addition of new cases. But question is if that would be ever needed.

Reply:
Likely won't have new cases. Just having FOO as an unknown status satisfies checking that unknown statuses will not be added to our output

Reply:
agreed with @pr-watson. Can't see a case where we would need to parametrize this.